src/Compilers/CSharp/Portable/Symbols/Attributes/WellKnownAttributeData/ParameterWellKnownAttributeData.cs

```diff
@@ -134,5 +137,22 @@ public bool HasEnumeratorCancellationAttribute
                 SetDataStored();
             }
         }
+
+        private HashSet<string> _notNullIfParameterNotNull;
+        public IImmutableSet<string> NotNullIfParameterNotNull
+        {
+            get
+            {
+                VerifySealed(expected: true);
+                return _notNullIfParameterNotNull?.ToImmutableHashSet() ?? ImmutableHashSet<string>.Empty;
```

Comment:
>_notNullIfParameterNotNull?.ToImmutableHashSet() [](start = 23, length = 48)

This will generate a new instance each time. Is there a reason to use a mutable `HashSet<string>` for the field rather than `ImmutableHashSet<string>`? The number of entries will be small, and only rarely > 1.